projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts

```diff
@@ -90,10 +91,11 @@ export class AgentsButtonComponent implements OnInit, OnDestroy {
       .subscribe(
         (agents) => {
           this._agents = agents;
+          this.cd.detectChanges();
           if (this.agents.length) {
             this._initialsByAgentId = this.agents.reduce((initials, agent) => {
               const words = agent.name.split(' ').filter((word) => !word.match(/[^a-zA-Z]+/g));
-              initials[agent.id] = `${words[0][0]}${words[1][0] || ''}`;
+              initials[agent.id] = words.length > 1 ? `${words[0][0]}${words[1][0]}` : `${words[0][0]}`;
```

Comment:
can you add unit tests for this case and second for checking if cd.detectChanges is called?